### PR TITLE
Bye bye HHVM.

### PR DESCRIPTION
**Pull request by derickr**

### CODE REVIEW

File: 201705250920-mongodb-hhvm.rst [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+.. articleMetaData::
+   :Where: London, UK
+   :Date: 2017-05-25 09:20 America/New_York
+   :Tags: blog, php, mongodb, php
+   :Short: hhvm-mongo
+
+At the start of 2015 we started work on an HHVM driver for MongoDB, as part
+of our project to renew our PHP driver. Back then, HHVM was in its ascendancy
+and outperforming PHP 5.6 two to one. Which such a huge performance difference
```
Comment (bjori) [inline 201705250920-mongodb-hhvm.rst]: `*With* such` ?

File: 201705250920-mongodb-hhvm.rst [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+it was clear that many users would be switching over. 
+
+Around the start of 2016 I wrote a series_ of blog_ posts_ when
+we released the brand new drivers for PHP and HHVM. However, by then, PHP 7
+had been released. PHP 7's performance was nearly equal to PHP 7 making it
```
Comment (bjori) [inline 201705250920-mongodb-hhvm.rst]: nearly equal to HHVM

File: 201705250920-mongodb-hhvm.rst [hunk 3]
```diff
@@ -0,0 +1,69 @@
+HHVM and MongoDB
+================
+
+.. articleMetaData::
+   :Where: London, UK
+   :Date: 2017-05-25 09:20 America/New_York
+   :Tags: blog, php, mongodb, php
+   :Short: hhvm-mongo
+
+At the start of 2015 we started work on an HHVM driver for MongoDB, as part
```
Comment (jmikola) [inline 201705250920-mongodb-hhvm.rst]: Replace "started" with "began" to avoid repetition.

File: 201705250920-mongodb-hhvm.rst [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+of our project to renew our PHP driver. Back then, HHVM was in its ascendancy
+and outperforming PHP 5.6 two to one. With such a huge performance difference
+it was clear that many users would be switching over.
```
Comment (jmikola) [inline 201705250920-mongodb-hhvm.rst]: "it was clear" or "it was reasonable to assume that"?

File: 201705250920-mongodb-hhvm.rst [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+   :Short: hhvm-mongo
+
+At the start of 2015 we started work on an HHVM driver for MongoDB, as part
+of our project to renew our PHP driver. Back then, HHVM was in its ascendancy
+and outperforming PHP 5.6 two to one. With such a huge performance difference
+it was clear that many users would be switching over. 
+
+Around the start of 2016 I wrote a series_ of blog_ posts_ when
+we released the brand new drivers for PHP and HHVM. However, by then, PHP 7
+had been released. PHP 7's performance is nearly equal to HHVM making it
```
Comment (jmikola) [inline 201705250920-mongodb-hhvm.rst]: I would refer to "PHP 7" as "PHP 7.0" here. Additionally, you may want to replace "nearly equal" with "on par with".

File: 201705250920-mongodb-hhvm.rst [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+less useful for users to move from PHP to HHVM. HHVM still has more strongly
```
Comment (jmikola) [inline 201705250920-mongodb-hhvm.rst]: "still has more" with "still offered more"

File: 201705250920-mongodb-hhvm.rst [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+typed PHP syntax through Hack, and some other support, but it's main
```
Comment (jmikola) [inline 201705250920-mongodb-hhvm.rst]: Replace "and some other support" with "and some other features"

File: 201705250920-mongodb-hhvm.rst [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+attraction: speed, was mostly gone.
```
Comment (jmikola) [inline 201705250920-mongodb-hhvm.rst]: Either replace ":" with "," or write "attraction of speed was mostly gone".

File: 201705250920-mongodb-hhvm.rst [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+
+.. _series: /new-drivers.html
+.. _blog: /new-drivers-part2.html
+.. _posts: /new-drivers-part3-cursor.html
+
+Writing an extension for PHP or for HHVM is very different. PHP extensions, of
```
Comment (jmikola) [inline 201705250920-mongodb-hhvm.rst]: "Writing an extension for HHVM is very different than doing so for PHP."

File: 201705250920-mongodb-hhvm.rst [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+which there are plenty, are written in C. HHVM extensions are written in C++,
+with very few third party extensions existing. At the same time, although
+PHP's APIs are not particularly well documented, there is a large group of
+people to ask for help. Writing an HHVM extension is a very different
```
Comment (jmikola) [inline 201705250920-mongodb-hhvm.rst]: After " there is a large group ofpeople to ask for help" I'd consider mentioning something about the internal APIs being clearly identified and stable across minor versions. This was a rather large annoyance with HHVM, as internal APIs would change between minor versions, although it wasn't clear if we were always using an internal API intended for extension authors.

File: 201705250920-mongodb-hhvm.rst [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+.. _poll: https://twitter.com/derickr/status/822027320005365761
+.. _`switch to HHVM`: https://twitter.com/doublecompile/status/822091880955596800
+
+With few users, frequent `API breaks`_, and `curious bugs`_ we came to the
+conclusion that supporting the HHVM driver for MongoDB no longer makes good
+use of our engineering time. With `Symfony 4 also dropping HHVM support`_ we
+have decided to discontinue the MongoDB driver for HHVM.
+
+If somebody wants to take over the GitHub repository, please drop me_ a line,
+and we will gladly hand it over.
```
Comment (jmikola) [inline 201705250920-mongodb-hhvm.rst]: This comes across as a bit flippant. How about:

> If anyone is interested in assuming ownership of the HHVM driver, please drop me a line and we can discuss the process in more detail.

File: 201705250920-mongodb-hhvm.rst [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+
+.. image:: /images/content/doublecompile.png
+   :title: https://twitter.com/doublecompile/status/822091880955596800
+
+.. _poll: https://twitter.com/derickr/status/822027320005365761
+.. _`switch to HHVM`: https://twitter.com/doublecompile/status/822091880955596800
+
+With few users, frequent `API breaks`_, and `curious bugs`_ we came to the
+conclusion that supporting the HHVM driver for MongoDB no longer makes good
+use of our engineering time. With `Symfony 4 also dropping HHVM support`_ we
```
Comment (jmikola) [inline 201705250920-mongodb-hhvm.rst]: If you'd like to cite Doctrine as well: https://github.com/doctrine/common/pull/785#issuecomment-294369979

**Comment by jmikola** [comment]
.

**Pull request merged**